libp2p/protocols/kademlia/dhttypes.nim

```diff
@@ -0,0 +1,62 @@
+#TODO: This file was mostly put together with the intent of using the typestate pattern
+# unfortunately, due to various design decisions, my assumption that the accessibility
+# controls needed for this to be a sane direction is not upheld. This file should be
+# considered a monument to my naivity, and to be dismantled
```

Comment:
This comment should be rephrased. It is fine for PR thread, but in code, self-flagelation does look poorly in a public codebase as it is just leaking vibe/mood. Replace with a TODO with specific actions